packages/mobile/src/send/saga.ts

```diff
@@ -169,12 +187,52 @@ function* sendPaymentOrInviteSaga({
   }
 }
 
+export function* validateRecipientAddressSaga({
+  userInputOfFullAddressOrLastFourDigits,
+  fullAddressValidationRequired,
+  recipient,
+}: ValidateRecipientAddressAction) {
+  Logger.debug(TAG, 'Starting Recipient Address Validation')
+  try {
+    if (!recipient.e164PhoneNumber) {
+      throw Error(`Invalid recipient type for Secure Send: ${recipient.kind}`)
+    }
+
+    const userAddress = yield select(currentAccountSelector)
+    const e164NumberToAddress = yield select(e164NumberToAddressSelector)
+    const { e164PhoneNumber } = recipient
+    const possibleRecievingAddresses = e164NumberToAddress[e164PhoneNumber]
+
+    // Should never happen since secure send is initiated due to there being several possible addresses
+    if (!possibleRecievingAddresses) {
+      throw Error('There are no possible recipient addresses to validate against')
+    }
+
+    const validatedAddress = validateAndReturnMatch(
+      userInputOfFullAddressOrLastFourDigits,
+      possibleRecievingAddresses,
+      userAddress,
+      fullAddressValidationRequired
+    )
+    yield put(validateRecipientAddressSuccess(e164PhoneNumber, validatedAddress))
+  } catch (error) {
+    Logger.error(TAG, 'validateRecipientAddressSaga/Address validation error: ', error)
+    yield put(showError(error.message))
```

Comment:
Note, if error.message isn't in `ErrorMessages` this may show garbage. See here for an example of how to handle that:
https://github.com/celo-org/celo-monorepo/blob/master/packages/mobile/src/identity/verification.ts#L198

Reply:
Nice, I like that